pkg/controller/migration/convert/convert_test.go

```diff
@@ -41,15 +41,9 @@ var _ = Describe("Parser", func() {
 	})
 
 	It("should detect an installation if one exists", func() {
-		c := fake.NewFakeClientWithScheme(scheme, &appsv1.DaemonSet{
-			ObjectMeta: v1.ObjectMeta{
-				Name:      "calico-node",
-				Namespace: "kube-system",
-			},
-		}, emptyKubeControllerSpec(), pool, emptyFelixConfig())
+		c := fake.NewFakeClientWithScheme(scheme, emptyNodeSpec(), emptyKubeControllerSpec(), pool, emptyFelixConfig())
 		err := Convert(ctx, c, &operatorv1.Installation{})
-		// though it will detect an install, it will be in the form of an incompatible-cluster error
-		Expect(err).To(BeAssignableToTypeOf(ErrIncompatibleCluster{}))
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
This is inverting the test. Why did that change?

Reply:
I intentionally flipped it. It's hard to explain, but I'll try to.

Part of the reason this is convoluted is because `Convert` used to return an error and an Installation resource. Nowadays, we pass in  a pointer to the Installation resource we've instantiated.

But if we go back in time to when it used to return both...

The test is basically checking that `Convert` doesn't return `nil, nil` when a calico-node daemonset exists. However, this test was written before we had helper methods for initializing valid components. So it passed a calico-node daemonset into the fake client which it knew would cause an ErrIncompatibleCluster because it's missing a ton of required spec fields, and then asserted that `Convert` is detecting that daemonset and not just returning `nil, nil`.

Reply:
Ok so since this test was changed to use the `emptyNodeSpec()` the node is now valid enough that an error is not expected.